src/pyodide/internal/workers.py

```diff
@@ -117,15 +139,25 @@ def __init__(
         )
         super().__init__(js_resp.url, js_resp)
 
+    @classmethod
+    def from_response(cls, body: Body, response: "Response") -> "Response":
+        b = body.js_object if isinstance(body, FormData) else body
+        result = cls.__new__(cls)
+        js_resp = js.Response.new(b, response.js_object)
+        super(Response, result).__init__(js_resp.url, js_resp)
```

Comment:
Can you explain the reasoning for this? Ideally we should switch to:
```suggestion
        js_resp = js.Response.new(b, response.js_object)
        return cls(js_resp.url, js_resp)
```
or `type(self)(js_resp.url, js_resp)` assuming we make it into an instance method. If this is different in some important way we'll need a comment. Particularly, I would expect `super(cls)` instead of `super(Response)` which skips the `__init__` functions of any subclasses.

Reply:
IIRC I was doing it this way because the constructor of `Response` doesn't match that of `pyodide.http.FetchResponse`: https://github.com/cloudflare/workerd/blob/a99aa9788cb8067304a868fb5d7677e0c8318947/src/pyodide/internal/workers.py#L120